tests/test_backoff.py

```diff
@@ -485,3 +485,104 @@ def test_backoff_conns_disconnect():
         for i, call in enumerate(c.stream.write.call_args_list):
             print "%d: %s" % (i, call)
         assert c.stream.write.call_args_list == [((arg,),) for arg in c.expected_args]
+
+
+def test_manual_backoff():
+    mock_ioloop = create_autospec(IOLoop)
+    r = _get_reader(mock_ioloop)
+    num_conns = 5
+    conns = []
+    for i in range(num_conns):
+        conn = _get_conn(r)
+        conn.expected_args = ['SUB test test\n', 'RDY 1\n']
+        conn.fails = 0
+        conns.append(conn)
+
+    conn = conns[0]
+
+    msg = _send_message(conn)
+
+    msg.trigger(event.FINISH, message=msg)
+    assert r.backoff_block is False
+    assert r.backoff_timer.get_interval() == 0
+
+    # calling resume() when not backed off should not raise any error
+    r.resume()
+
+    r.backoff()
+
+    # calling backup() when already backed off should not raise any error
+    r.backoff()
+
+    send_args, send_kwargs = conn.stream.write.call_args
+    assert r.backoff_block is True
+    assert send_args[0] == 'RDY 0\n'
+    assert r.backoff_timer.get_interval() == 0, 'manual backoff must not increase backoff timer interval'
+
+    call_arg_count_map = get_call_arg_count_map(conns)
+
+    r.resume()
+
+    assert r.backoff_block is False
+    assert get_ready_counts(conns, call_arg_count_map) == 5, 'resume from manual backoff must send full ready count'
+
+    now = time.time()
+    backoff_duration = 99
+    r.backoff(backoff_duration)
+
+    assert r.backoff_block is True
+    assert mock_ioloop.add_timeout.called
+    timeout_args, timeout_kwargs = mock_ioloop.add_timeout.call_args
+    assert are_floats_close(timeout_args[0], now + backoff_duration), \
```

Comment:
Now is seconds since unix epoch, which is a pretty big number. are_floats_close() might not be the best way to test timing tolerance.

are_floats_close() currently allows a "relative tolerance" of 1e-9 which comes out to about 1.45 seconds for time_t right now. That's not much, but consider how odd it is that the tolerance allowed will increase over the months and years. It would be simpler to compare to a specific time tolerance (and are_floats_close() could be removed)